pkg/dyninst/decode/doc.go

```diff
@@ -8,4 +8,70 @@
 // Package decode is responsible for decoding raw event bytes from dynamic
 // instrumentation's ebpf ringbuffer into JSON to be uploaded to the datadog
 // logs backend.
+//
+// JSON is written directly to the decoder's output stream for efficiency.
+// An example of the JSON output is:
+//
+// ```json
+//
+//	{
+//	    "service": "sample",
+//	    "ddsource": "dd_debugger",
+//	    "ddtags": "",
+//	    "logger": {
+//	        "name": "",
+//	        "method": ""
+//	    },
+//	    "debugger": {
+//	        "snapshot": {
+//	            "id": "fdf8b824-ecbc-47ea-bab0-60c70e12dff8",
+//	            "timestamp": 1750927066975,
+//	            "language": "go",
+//	            "stack": [
+//	                {
+//	                    "function": "main.testSingleRune",
+//	                    "file": "/home/vagrant/datadog-agent/pkg/dyninst/testprogs/progs/sample/basics.go",
+//	                    "line": 15
+//	                },
+//	                {
+//	                    "function": "main.main",
+//	                    "file": "/home/vagrant/datadog-agent/pkg/dyninst/testprogs/progs/sample/main.go",
+//	                    "line": 24
+//	                },
+//	                {
+//	                    "function": "main.main",
+//	                    "file": "/home/vagrant/datadog-agent/pkg/dyninst/testprogs/progs/sample/main.go",
+//	                    "line": 24
+//	                },
+//	                {
+//	                    "function": "runtime.main",
+//	                    "file": "/usr/local/go/src/runtime/proc.go",
+//	                    "line": 283
+//	                },
+//	                {
+//	                    "function": "runtime.goexit",
+//	                    "file": "/usr/local/go/src/runtime/asm_arm64.s",
+//	                    "line": 1223
+//	                }
+//	            ],
+//	            "probe": {
+//	                "location": {
+//	                    "method": "main.testSingleRune"
+//	                }
+//	            },
+//	            "captures": {
+//	                "entry": {
+//	                    "arguments": {
+//	                        "x": {
+//	                            "type": "int32",
+//	                            "value": 1
+//	                        }
+//	                    }
+//	                }
+//	            }
+//	        }
+//	    }
+//	}
+//
+// ```
```

Comment:
In go doc comments you just need to indent for it to be block quoted and pre-formatted -- the \`\`\` doesn't do anything and ultimately renders in quite an ugly way:

> A pair of consecutive backticks (` U+0060) is interpreted as a Unicode left quote (“ U+201C), and a pair of consecutive single quotes (' U+0027) is interpreted as a Unicode right quote (” U+201D).

https://tip.golang.org/doc/comment#paragraphs